test/validators/movie.test.js

```diff
@@ -44,12 +46,78 @@ describe('movie validator', () => {
         title: 'foo',
         release_year: 12345
       };
-      const result = Joi.validate(payload, MovieValidator);
+      const result = Joi.validate(payload, MovieValidator.payload);
 
       expect(result.error.details[0].path).to.eql('release_year');
       expect(result.error.details[0].type).to.eql('number.max');
     });
 
   });
 
+  describe('release_year_start', () => {
+
+    it('is after 1878', () => {
+      const query = {
+        title: 'poop',
+        release_year_start: 1776
+      };
+      const result = Joi.validate(query, MovieValidator.query);
+
+      expect(result.error.details[0].path).to.eql('release_year_start');
+      expect(result.error.details[0].type).to.eql('number.min');
+    });
+
+    it('is limited to 4 digits', () => {
+      const query = {
+        title: 'foo',
+        release_year_start: 12345
+      };
+      const result = Joi.validate(query, MovieValidator.query);
+
+      expect(result.error.details[0].path).to.eql('release_year_start');
+      expect(result.error.details[0].type).to.eql('number.max');
+    });
+
+  });
+
+  describe('release_year_end', () => {
+
+    it('is after 1878', () => {
+      const query = {
+        title: 'poop',
+        release_year_end: 1776
+      };
+      const result = Joi.validate(query, MovieValidator.query);
+
+      expect(result.error.details[0].path).to.eql('release_year_end');
+      expect(result.error.details[0].type).to.eql('number.min');
+    });
+
+    it('is limited to 4 digits', () => {
+      const query = {
+        title: 'foo',
+        release_year_end: 12345
+      };
+      const result = Joi.validate(query, MovieValidator.query);
+
+      expect(result.error.details[0].path).to.eql('release_year_end');
+      expect(result.error.details[0].type).to.eql('number.max');
+    });
+
+  });
+  
+  describe('fuzzy_title', () => {
+ 
+    it('is less than 255 characters', () => {
+      const query = { 
+        fuzzy_title: 'jx'.repeat(260) 
```

Comment:
`'jx'.repeat(260)` will be of length 520. Ideally we want to check the smallest increment after the maximum allowed value (in this case 256).